### PR TITLE
DPC-1208 Post Group

### PR DESCRIPTION
**Pull request by whuang85**
### Fixes [DPC-1208](https://jira.cms.gov/browse/DPC-1208)

In V2, we want to be able to export Groups without having to post a practitioner and patients separately, this PR allows a user to Post a Group with the practitioner info for each patient. This is modeled after the DaVinci [structure definition ](http://hl7.org/fhir/us/davinci-atr/StructureDefinition-atr-group.html).  It is assumed that each practitioner that is associated with the patient is also all under the same org umbrella.

### Proposed Changes

* Add ability to post a group
* Filter out extraneous group data
* Add HTML Sanitize filter to sanitize request bodies

### Security Implications

- [x] new software dependencies

see go.mod

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### CODE REVIEW

File: dpc-go/dpc-api/src/apitest/helper.go [hunk 1]
```diff
@@ -101,186 +101,100 @@ const Orgjson = `{
...
 
-// Practitionerjson is a practitioner json string for testing purposes
-const Practitionerjson = `{
-  "resourceType": "Practitioner",
-  "id": "f001",
-  "text": {
-    "status": "generated",
-    "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\"><p><b>Generated Narrative with Details</b></p><p><b>id</b>: f001</p><p><b>identifier</b>: 938273695 (OFFICIAL), 129IDH4OP733 (USUAL)</p><p><b>name</b>: Eric van den broek (OFFICIAL)</p><p><b>telecom</b>: ph: 0205568263(WORK), E.M.vandenbroek@bmc.nl(WORK), fax: 0205664440(WORK)</p><p><b>address</b>: Galapagosweg 91 Den Burg 9105 PZ NLD (WORK)</p><p><b>gender</b>: male</p><p><b>birthDate</b>: 07/12/1975</p></div>"
+// Groupjson is a group json string for testing purposes
+const Groupjson = `
```
Comment (dhgreene) [inline dpc-go/dpc-api/src/apitest/helper.go:105]: This example with it's "extra" information is fine, but we also need a clear sample somewhere showing what we're actually using.

Comment (whuang85) [reply dpc-go/dpc-api/src/apitest/helper.go:105]: @dhgreene wouldn't that be where our smoke tests and e2e tests come in to clarify the structure?

Comment (dhgreene) [reply dpc-go/dpc-api/src/apitest/helper.go:105]: Care to add one of those?  😉 (Don't worry--ticket coming!)  The immediate issue is that, beyond the code in this PR, we're evaluating the data structure.  Does it really comply with the DaVinci IG?  Are there any "gotchas" making our own IG that derives from it?  Is there a better way to solve our need?  To answer those questions we need that minimal sample.  I'd be ok with dropping it in the PR description.

Comment (whuang85) [reply dpc-go/dpc-api/src/apitest/helper.go:105]: @dhgreene I just added a filtered group json after it passes through the fhir_filter.  That should basically be a happy path request

File: dpc-go/dpc-api/src/model/fhir.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	Name           *string                    `json:"name,omitempty"`
+	Characteristic []fhir.GroupCharacteristic `json:"characteristic,omitempty"`
+	Member         []GroupMember              `json:"member,omitempty"`
+	ResourceType
+}
+
+// GroupMember is a struct that represent the filtered down fhir.GroupMember
+type GroupMember struct {
+	Entity    *fhir.Reference `json:"entity"`
+	Period    *fhir.Period    `json:"period,omitempty"`
```
[the remark this answers is not in the document]
Comment (whuang85) [reply dpc-go/dpc-api/src/model/fhir.go:32]: Nope, currently the Java stuff does so just included it. But after discussing how long our groups will live, these will probably not be needed

[the remark this answers is not in the document]
Comment (dhgreene) [reply dpc-go/dpc-api/src/model/fhir.go:32]: @whuang85 Shall we delete them, then?  I'm ready to approve at that point.

**Review by dhgreene — APPROVED** [review]
💪

**Pull request merged**